### PR TITLE
(from AES) Implement Deployment and ConfigMap report in Agent

### PR DESCRIPTION
**Pull request by leoluz**
## Description
This PR adds ConfigMaps and Deployments to the report sent to Ambassador Cloud.

## Testing
Added unit-tests to the new core-store and manually ran e2e test deploying in my personal kubeception cluster and analyzing the report sent to Ambassador Agent

## Checklist

 - [ ] I made sure to update `CHANGELOG.md`.
   
   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations
 
 - [ ] This is unlikely to impact how Ambassador performs at scale.
 
   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability
 
 - [x] My change is adequately tested.
 
   Remember when considering testing:
    + LEGACY MODE TESTS DO NOT RUN FOR EVERY PR. If your change is affected by legacy mode, you need
      to run legacy-mode tests manually (set `AMBASSADOR_LEGACY_MODE=true` and run the tests).
      (This will be fixed soon.)
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points
 
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

### CODE REVIEW

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -492,9 +500,19 @@ func (a *Agent) ProcessSnapshot(ctx context.Context, snapshot *snapshotTypes.Sna
 	}
 
 	if snapshot.Kubernetes != nil {
-		if a.podStore != nil {
-			snapshot.Kubernetes.Pods = a.podStore.GetPodsForServices(snapshot.Kubernetes.Services)
-			dlog.Debugf(ctx, "Found %d pods and %d services", len(snapshot.Kubernetes.Pods), len(snapshot.Kubernetes.Services))
+		if a.coreStore != nil {
+			if a.coreStore.podStore != nil {
+				snapshot.Kubernetes.Pods = a.coreStore.podStore.StateOfWorld()
+				dlog.Infof(ctx, "Found %d pods", len(snapshot.Kubernetes.Pods))
```
Comment (acookin) [inline pkg/agent/agent.go:506]: maybe add this as a debug line? cuz that'll be a lotta logs. same for the below log lines

File: pkg/agent/corestore.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	for _, v := range store.sotw {
+		deployments = append(deployments, v)
+	}
+	return deployments
+}
+
+// allowedNamespace will check if resources from the given namespace
+// should be reported to Ambassador Cloud.
+func allowedNamespace(namespace string) bool {
+	if namespace != NamespaceKubeSystem {
```
Comment (acookin) [inline pkg/agent/corestore.go:123]: but what if the user is running services in kube system?

File: pkg/agent/corestore.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	}
+}
+
+// NewPodStore will create a new podStore filtering out undesired resources.
+func NewPodStore(pods []*kates.Pod) *podStore {
+	sotw := make(map[string]*kates.Pod)
+	store := &podStore{sotw: sotw}
+
+	for _, pod := range pods {
+		if allowedNamespace(pod.GetNamespace()) && pod.Status.Phase != v1.PodSucceeded {
```
Comment (acookin) [inline pkg/agent/corestore.go:57]: does it also make sense to filter out failed here?

**Review by acookin — CHANGES_REQUESTED** [review]
overall just looks good. definitely want to fix that log line, but i'm willing to budge on the other feedback

File: pkg/agent/agent.go (hunk 1 above)
Comment (leoluz) [reply pkg/agent/agent.go]: Definitely. 
Done.

File: pkg/agent/corestore.go (hunk 2 above)
Comment (leoluz) [reply pkg/agent/corestore.go:123]: I had a discussion with Alex and Max about the rules to apply while reporting to Ambassador Cloud. No one should be deploying their services in the `kube-system` namespace. I believe that the amount of people that will red flag this is higher than the ones the actually deploy services in `kube-system`

File: pkg/agent/corestore.go (hunk 3 above)
Comment (leoluz) [reply pkg/agent/corestore.go:57]: We also spoke about this and we found useful to send pods that are in failed state so we could inspect them and provide useful info to our users on how to address those issues.

**Review by acookin — APPROVED** [review]

**Pull request merged**